server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -63,6 +64,29 @@
         PARSER.declareInt(AutoDateHistogramAggregationBuilder::setNumBuckets, NUM_BUCKETS_FIELD);
     }
 
+    /**
+     *
+     * Build roundings, computed dynamically as roundings are time zone dependent.
+     * The current implementation probably should not be invoked in a tight loop.
+     * @return Array of RoundingInfo
+     */
+    RoundingInfo[] buildRoundings() {
```

Comment:
Another thing we could do here (to avoid having to create an instance of the builder in the `InternalAutoDateHistogramTests`) is to make the signature here: `static RoundingInfo[] buildRoundings(DateTimeZone)` and then pass the time zone into it the method both in `innerBuild()` and in `InternalAutoDateHistogramTests`. We will also need to make `createRounding(DateTimeUnit, DateTimeZone)` static as well but thats fine I think.